tajo-core/src/main/java/org/apache/tajo/engine/function/FunctionLoader.java

```diff
@@ -299,4 +284,38 @@ private static StaticMethodInvocationDesc extractStaticMethodInvocation(Method m
 
     return sqlFuncs;
   }
+
+  public static Collection<FunctionDesc> loadFunctions(TajoConf conf) throws IOException, AmbiguousFunctionException {
+    List<FunctionDesc> functionList = new ArrayList<>(loadBuiltinFunctions().values());
+    List<FunctionDesc> udfs = loadUserDefinedFunctions(conf);
+
+    return mergeFunctionLists(functionList, udfs);
+  }
+
+  @SafeVarargs
+  static Collection<FunctionDesc> mergeFunctionLists(List<FunctionDesc> ... functionLists)
+      throws AmbiguousFunctionException {
+
+    Map<Integer, FunctionDesc> funcMap = new HashMap<>();
+    List<FunctionDesc> baseFuncList = functionLists[0];
+
+    // Build a map with a first list
+    for (FunctionDesc desc: baseFuncList) {
+      funcMap.put(desc.hashCodeWithoutType(), desc);
+    }
+
+    // Check duplicates for other function lists(should be UDFs practically)
+    for (int i=1; i<functionLists.length; i++) {
+      for (FunctionDesc desc: functionLists[i]) {
+        if (funcMap.containsKey(desc.hashCodeWithoutType())) {
```

Comment:
As you know, the equality check with hash code involves a possibility of false positive. I know it is very rare, but we need to check the real equality for valid operation.